### ISSUE TITLE
Audio pitching info

### ISSUE THREAD

**Issue by geoku** [IceDBorn/pipewire-screenaudio#22]
For me Firefox 117 didn't fix anything in relation to audio pitching. It behaves absolutely the same as before.

But today I've made a discovery: when my stream is already going and the receiving end is hearing heavily distorted audio, it turns out it can be fixed by going to the Voice & Video section of Discord's settings and making any change to the "Input Device" dropdown. For example, I have my Razer mic selected there, so I switch to anything else. As soon as I do this, audio pitching is immediately fixed on the fly. Then I switch the dropdown back to my Razer mic, and the audio stays fixed.

TL;DR steps:
1. Share a node using the extension and start your stream (viewer hears distorted audio);
2. Go to Voice & Video settings in Discord;
3. Change "Input Device" from your mic to anything else;
4. Audio is immediately fixed;
5. Change "Input Device" back to your mic.

**It works even in Firefox 116. Firefox 117 didn't change anything.**

**Comment by IceDBorn** [comment]
This works on Jitsi too. A way to fix this would be removing the selected microphone from the call and then adding it back.

**Comment by IceDBorn** [comment]
https://github.com/IceDBorn/pipewire-screenaudio/commit/952408dd513b47c20216f4a3cc7c108441eba58a

**Comment by IceDBorn** [comment]
For anyone interested, a mozilla employee posted this [WIP build](https://bugzilla.mozilla.org/show_bug.cgi?id=1844181#c17) that fixes the audio pitching bug

**Comment by illuminor** [comment]
Follow up to the initial post: turns out, if you do the steps described, your mic's audio becomes distorted for the listener instead of the stream audio. I didn't notice this initially, because I was testing with myself and didn't try speaking into the mic. But it was pointed out to me when I tried this in a call with another person. Ultimately, it seems like the last sound source connected to the call will have distorted audio.

**Comment by Xefir** [comment]
I found a way to not have any audio distortion, but you have to strictly respect this time line of events : 

- Your browser must not have started any call attempt. If so, restart your browser.
- Run you game or the window you want the audio to be shared
- Go to the webpage you want to share the game or window
- In the firefox extension, share the game or window (yes, before starting any call at all)
- In the website, pre-select your mic (the real one you use to talk to) and force the app to stay to this mic (Discord have this disgusting habbit to ask you every time to change when a new mic is added, do not let it do that).
- Start your call
- Start your share screen (again some apps will want to change mic, refuse to let it happens)

Now, the screen, mic and share screen audio will be pitch perfect and if you share the same app on the same website, even if you stop and restart the call after.
But, you'll have to restart your browser and do the same if you want another window to share or a another website.

**Comment by aeongdesu** [comment]
[let's go! 🚀](https://bugzilla.mozilla.org/show_bug.cgi?id=1844181#c70)

![](https://github.com/IceDBorn/pipewire-screenaudio/assets/50764666/fb2ac0df-1945-412f-916b-5041af08b372)

**Comment by IceDBorn** [comment]
We would be glad if anyone of you can test this out. You can just download the nightly version of Firefox.

**Comment by aeongdesu** [comment]
@IceDBorn yeah works fine on my machine ✌️


https://github.com/IceDBorn/pipewire-screenaudio/assets/50764666/494e94b1-2804-4bc9-a2d7-bfbe7b231303



```shell
OS: EndeavourOS Linux x86_64 
Kernel: 6.5.6-arch2-1
Shell: bash 5.1.16 
Resolution: 1920x1080 
DE: Plasma 5.27.8 
WM: KWin
CPU: Intel i5-9400F (6) @ 4.100GHz 
GPU: NVIDIA GeForce RTX 2060 Rev. A 
Memory: 3934MiB / 15938MiB 

[nyan@arch firefox]$ ./firefox -version
Mozilla Firefox 120.0a1
```

**Comment by kheiken** [comment]
Same for me, the latest nightly fixes the pitch issues. Also when using screenaudio and a microphone, enabled in any order and if disabled and re-enabled (which was an issue in early versions of the fix/workarounds).

Combined with #27 I can now screenshare on Debian 12 using e.g. Discord with microphone being on at the same time and pitch is fine for both.

Thanks to you all for this great project!

**Comment by IceDBorn** [comment]
We need testers for Firefox Beta!

**Comment by jim3692** [comment]
Looks like NixOS Unstable is still on Beta 119. On Arch the Beta 120 is released in AUR. We need to test on 120.

**Comment by pallaswept** [comment]
I've been watching this repo and waiting for 120 to land here, and it just hit opensuse tumbleweed. Will test as soon as I wake up.

**Comment by pallaswept** [comment]
No problems with the audio on the receiving side in 120 here. I never experienced this bug (I didn't use the plugin until now) so I'm not sure what I should be listening out for, but everything sounds fine. There is definitely no pitch shifting or distortion.
